src/Microsoft.OData.Core/JsonLight/ODataJsonLightInputContext.cs

```diff
@@ -636,6 +647,12 @@ private static IJsonReader CreateJsonReader(IServiceProvider container, TextRead
             var jsonReader = jsonReaderFactory.CreateJsonReader(textReader, isIeee754Compatible);
             Debug.Assert(jsonReader != null, "jsonWriter != null");
 
+            JsonReader odataJsonReader = jsonReader as JsonReader;
+            if (odataJsonReader != null)
+            {
+                odataJsonReader.ArrayPool = container.GetService<ICharArrayPool>();

```

Comment:
Can you explain how this is set? I don't see this being set even in the configuration for tests.